cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -828,8 +837,17 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
                  txOuts = TxOut changeaddr balance TxOutDatumHashNone
                         : txOuts txbodycontent
                }
+    let finalFee = evaluateTransactionFee pparams txbody3 nkeys 0
+        diff = finalFee - fee
+        finalBalance = lovelaceToTxOutValue $ txOutValueToLovelace balance - diff
 
-    return txbody3
+    txbody4 <- first TxBodyError $ -- TODO: impossible to fail now
+               makeTransactionBody txbodycontent1 {
+                 txFee  = TxFeeExplicit explicitTxFees finalFee,
+                 txOuts = TxOut changeaddr finalBalance TxOutDatumHashNone
+                        : txOuts txbodycontent
+               }
+    return txbody4
```

Comment:
So the reason for this extra iteration, I think, is that when we calculate the fee we were doing it with a change output of zero, which is smaller than the final output.

I think we could do this without an extra round if we use a bigger dummy value for the change output when we compute the fees.

Reply:
How big should the dummy value be?